### PR TITLE
Only post changed fields on Save.

### PR DESCRIPTION
**Pull request by FodT**
This PR changes the behaviour of `properties` to only show changed fields. The Entity `__repr__()` function uses `_init_properties` and overrides them with any changed values from `properties`. On save, only the fields in `properties` are posted, along with the version. 

This has two advantages: 
 * minimises POST body size
 * since only the changed fields are now being posted, we don't need the pacing migration code in Campaigns or Strategies. 

This PR also removes some deprecated functions, along with basic auth.

### CODE REVIEW

File: terminalone/connection.py [hunk 1]
```diff
@@ -119,14 +119,6 @@ def _auth_session_id(self, session_id, api_key, expires=None):
         )
         self._check_session()
 
-    def _auth_basic(self, username, password, api_key):
```
Comment (fsargent) [inline terminalone/connection.py]: Yayyyyyy

File: terminalone/entity.py [hunk 2]
```diff
@@ -38,44 +38,29 @@ def __init__(self, session, properties=None, **kwargs):
...
         return '{cname}({props})'.format(
             cname=type(self).__name__,
             props=', '.join(
                 '{key}={value!r}'.format(key=key, value=value)
-                for key, value in six.iteritems(self.properties)
+                for key, value in six.iteritems(properties)
             )
         )
 
-    def __getitem__(self, attribute):
```
Comment (fsargent) [inline terminalone/entity.py]: YAY!

File: terminalone/entity.py [hunk 3]
```diff
@@ -149,13 +126,15 @@ def _validate_form_post(self, data):
 
         If attribute should not be sent, remove it from the body.
         """
-        update = 'id' in self.properties
-        if 'version' not in data and update:
+        entity_id = self._init_properties.get('id', False)
```
Comment (pswaminathan) [inline terminalone/entity.py]: Why is this being changed to have an ID? It seems like all uses of it are still checking that it exists the way it was before.

**Comment by FodT** [comment]
Resolves #129

**Pull request merged**